app/views/events/index.html.erb

```diff
@@ -12,62 +12,92 @@
   <thead>
   <tr>
     <% [:name, :game_mode, :max_teams, :deadline, :startdate, :enddate].each do |attribute| %>
-        <th><%= model_class.human_attribute_name(attribute) %></th>
+      <th><%= model_class.human_attribute_name(attribute) %></th>
     <% end %>
-    <th><%= t '.actions', :default => t("helpers.actions") %></th>
+    <th><%= t '.actions', default: t("helpers.actions") %></th>
   </tr>
   </thead>
   <tbody>
   <% @events.each do |event| %>
-      <tr>
-        <td><%= link_to event.name, event_path(event) %></td>
-        <td><%= event.human_game_mode %></td>
-        <td><%= event.max_teams %></td>
-        <td><%= event.deadline %></td>
-        <td><%= event.startdate %></td>
-        <td><%= event.enddate %></td>
-        <td>
-          <%= link_to t('.show', :default => t("helpers.links.show")),
-                      event_path(event), :class => 'btn btn-default btn-xs' %>
+    <tr>
+      <td><%= link_to event.name, event_path(event) %></td>
+      <td><%= event.human_game_mode %></td>
+      <td><%= event.max_teams %></td>
+      <td><%= event.deadline %></td>
+      <td><%= event.startdate %></td>
+      <td><%= event.enddate %></td>
+      <td>
+        <%= link_to t('.show', default: t("helpers.links.show")),
+                    event_path(event), class: 'btn btn-default btn-xs' %>
 
-          <% if can? :leave, event %>
-            <%= link_to t('.leave', :default => t("helpers.links.leave")),
+        <% if can? :leave, event %>
+          <% if event.single? %>
+            <%= link_to t('.leave', default: t("helpers.links.leave")),
                         leave_event_path(event),
-                        :method => :put,
-                        :class => 'btn btn-xs btn-default',
+                        method: :put,
+                        class: 'btn btn-xs btn-default',
+                        id: :leave_event_button %>
+          <% else %>
+            <%= link_to t('.leave', default: t("helpers.links.leave")),
+                        leave_event_path(event),
+                        method: :put,
+                        class: 'btn btn-xs btn-default',
+                        disabled: (not event.ownes_participating_teams?(current_user)),
```

Comment:
The only point in which the if and else blocks differ is the value of "disabled". The change in control flow might be moved to that point. Also, since all single player events know what teams are and for every single player team the user is also the owner, the question whether the user owns a team would always return true for single player anyway.